app/controllers/CyaMaintainProfileController.scala

```diff
@@ -137,8 +135,44 @@ class CyaMaintainProfileController @Inject() (
             } yield Redirect(navigator.nextPage(CyaMaintainProfilePage, CheckMode, request.userAnswers))
           }
         case Left(errors)       =>
-          Future.successful(logErrorsAndContinue(errorMessage, continueUrl, errors))
+          Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
       }
     }
 
+  def onPageLoadNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
+        TraderProfile.buildNirms(request.userAnswers, request.eori, traderProfile) match {
```

Comment:
i could be super picky and say we don't actually need the buildNirms bit, we just need the getOptionallyRemovedPage part because then that saves you having to get the trader profile

Reply:
getOptionallyRemovedPage is on TraderProfile, I could make that public?

Reply:
Done :)